### ISSUE TITLE
why I use conv2d with padding='VALID' is more slow than SAME?

### ISSUE THREAD

**Issue by brantbzhang** [tensorflow/tensorflow#1545]
I use cnn for text classification.Document max len:200;Embedding size:120
When I use conv2d with padding='VALID'.  127.6 examples/sec
When I use conv2d with padding='SAME'.  333.3 examples/sec

def cnn_model(X,prefix,is_train=False):
    with tf.variable_scope(prefix):
        with tf.device('/gpu:0'):
            embeddings = tf.Variable(
                tf.random_uniform([WORDS_NUM, EMBEDDING_SIZE], -1.0, 1.0))
            embed = tf.nn.embedding_lookup(embeddings, X)
        word_vectors = tf.expand_dims(embed, 3)
        with tf.variable_scope('CNN_Layer1'):
            conv1=skflow.ops.conv2d(word_vectors,64,(2,EMBEDDING_SIZE),(1,1,1,1),padding='VALID')
            conv1=tf.nn.relu(conv1)
            pool1=tf.nn.max_pool(conv1,ksize=[1,2,1,1],strides=[1,2,1,1],padding='SAME')
        net=tf.reshape(pool1,[BATCH_SIZE,-1])
        if is_train:
            net=tf.nn.dropout(net, 0.7)
        net = slim.ops.fc(net, NUM_LABELS, activation=None, scope='fc8')
        return net

**Comment by mrry** [comment]
Assigning @zheng-xq as the most expert on GPU performance issues.

It would help to know which version of the code you're using, because @zheng-xq recently made some substantial performance improvements, so these numbers might be different at HEAD.

**Comment by brantbzhang** [comment]
commit 69a5419a61a37a2674b4e6573e9a51828c5d1e9f
Merge: b9e9c9a bc8d5fe
Author: Martin Wicke <martin.wicke@gmail.com>
Date:   Mon Mar 14 21:59:58 2016 -0700

    Merge pull request #1507 from chemelnucfin/quickfix

    quick python3 fix


Thanks!

**Comment by brantbzhang** [comment]
I find that:
when I set conv2d kernel size with [2,EMBEDDING_SIZE,1,64] or [1,EMBEDDING_SIZE,1,64],it is very slow(113.2 examples/sec).
But when I set kernel size with [3,EMBEDDING_SIZE,1,64] ,it's 11502.3 examples/sec,100 times faster.

All padding with VALID.

**Comment by zheng-xq** [comment]
@brantbzhang, could you try the latest TensorFlow, which enables auto-tune on conv and see if that is faster?

**Comment by aselle** [comment]
Automatically closing because there was no response. Please reopen if it is still an issue.
